test/cql-pytest/test_using_timestamp.py

```diff
@@ -90,3 +90,79 @@ def test_key_writetime(cql, table1):
         cql.execute(f'SELECT writetime(k) FROM {table1}')
     with pytest.raises(InvalidRequest, match='PRIMARY KEY part k'):
         cql.execute(f'SELECT ttl(k) FROM {table1}')
+
+# Reproducer for https://github.com/scylladb/scylladb/issues/14182
+def test_rewrite_using_same_timestamp(cql, table1):
+    table = table1
+    k = unique_key_int()
+    ts = 1000
+    ttl1 = 1
+    ttl2 = 1
+    values = [[1, 1], [1, 2], [2, 1]]
+    errors = []
+    for i in range(3):
+        v1, v2 = values[i]
+
+        # sync with wall-clock on exact second so that expiration won't cross the whole-second boundary
+        t1 = time.time()
+        while t1 - int(t1) >= 0.9:
+            time.sleep(1 - (t1 - int(t1)))
+            t1 = time.time()
```

Comment:
I think this is risky. On a very slow machine it can do weird things. Why is this even necessary?

Reply:
ttl is quantized in full second resolution.
If the two consecutive inserts cross the full second boundary they will not have the same expiry anymore and this makes the test flaky (I actually reproduced that in debug mode with a millisecond to go before the full-second mark). 10 milliseconds margin proved to be reliable enough on my local machine  in debug mode.